### DISCUSSION TITLE
Serving static HTML files without extension

### DISCUSSION THREAD

**Discussion by ekawatani** [vercel/community#1486] [Help] [answered]
Hi,

I have some old HTML files that I need to serve with vercel + next.js, but when I put HTML files under `public`, they are only accessible via `/filename.html`, but I want them to be accessible via `/filename` just like other pages in the pages directory.

Is there a way to do this? I read that running a custom server is the way to go. But, as far as I know, this is only possible for the `/api` routes.

**Comment by wiiiimm** [comment]
@ekawatani You can setup routes using vercel.json. Please checkout https://vercel.com/docs/configuration#routes for more info.

**Reply by ekawatani** [reply]
Thanks, I'll give it a try. It sounds promising.

**Comment by domdomegg** [comment, marked as answer]
As an alternative to people coming to this later, the config option in vercel.json you probably want is [`cleanUrls`](https://vercel.com/docs/concepts/projects/project-configuration#cleanurls):

```json
{
  "cleanUrls": true
}
```

From the docs: 'For example, a static file named about.html will be served when visiting the /about path. Visiting /about.html will redirect to /about.'

**Reply by amyegan** [reply]
Thanks for sharing this updated answer!
